core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -161,13 +161,18 @@ private Cursor newCursor(boolean sorted) {
     private int i = -1;
     private long base = 0;
     private int[] table;
+    private final Integer[] sortedRef;
```

Comment:
Not only memory but performance and creation of unnecessary objects. We need to minimize pressure on garbage collection

Reply:
The previous implementation is creating a clone of the tagSets (`tagSets.clone();`) and sorts the tagsets so the iterator calling the cursor will iterate over the sorted list that's needed for the metric registry to group metrics of the same tagset. 
The current implementation avoids cloning the tagSets (and sorting it) and instead sorts just the Integer array that contains the indices of the tagSets (see https://github.com/ultrabrew/metrics/pull/28/files#diff-fa8eef85c6812afee9ae6b074bdc7cbfR47-R48). When the metric registry needs to iterate over the tagSets in sorted order, it simply goes over the sorted indices array (the `Integer[] `in this case). Now, I wanted to use the `Arrays.sort `functionality since it's the most efficient sort library that optimizes and runs various algorithms depending on the array length. 

The other reason to use Object array instead of the native array is that since I wanted to use the `Arrays.sort`, it needs `java.util.Comparator` and the Comparator can only work with Objects.

Reply:
What if tagsets changes while you hold the cursor to it? Previous implementation did not have this problem as it has a copy of it while iterating. Or has this been addressed already someway?

Reply:
I would still prefer using primitive array instead of object array of objects. Lots of garbage. Cant search now but pretty sure there is some quicksort for primitive arrays also